src/vmm/src/arm_templates/mod.rs

```diff
@@ -0,0 +1,54 @@
+use core::ops::Deref;
+use std::collections::BTreeMap;
+
+use kvm_ioctls::VcpuFd;
+
+pub mod g3g2;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Error getting register {0}: {1}")]
+    GetReg(u64, kvm_ioctls::Error),
+    #[error("Error setting register {0}: {1}")]
+    SetReg(u64, kvm_ioctls::Error),
+}
+
+pub struct RegisterValue {
+    // We zero bits with zero mask and set with values mask
+    // (register & mask) | value
+    mask: u128,
+    value: u128,
+}
+
+pub struct CpuTemplateAarch64(BTreeMap<u64, RegisterValue>);
+
+impl Deref for CpuTemplateAarch64 {
+    type Target = BTreeMap<u64, RegisterValue>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl CpuTemplateAarch64 {
+    pub fn registers_ids(&self) -> Vec<u64> {
+        self.keys().copied().collect()
+    }
+}
+
+pub fn apply_template(
+    vcpu: &VcpuFd,
+    template: &CpuTemplateAarch64,
+) -> std::result::Result<(), Error> {
+    for (reg_id, reg_value) in template.iter() {
+        let value = {
+            let curr_value = vcpu
+                .get_one_reg(*reg_id)
+                .map_err(|error| Error::GetReg(*reg_id, error))?;
+            (curr_value & reg_value.mask) | reg_value.value
```

Comment:
still think we have to align with @mattschlebusch 's interpretation (clear+set vs mask+value semantics).